nrf_rpc/include/nrf_rpc.h

```diff
@@ -357,6 +371,42 @@ void nrf_rpc_set_bound_handler(nrf_rpc_group_bound_handler_t bound_handler);
  */
 int nrf_rpc_init(nrf_rpc_err_handler_t err_handler);
 
+
+/** @brief Registers the cleanup handler.
+ *
+ * Calling this adds the handler to the list. The handler will be called automatically during @ref nrf_rpc_stop
+ * if the cleanup parameter is set to true.
+ * This API is used to clean up any custom states set during RPC operations, for example clean up callbacks set.
+ *
+ * @param handler  Pointer to the structure storing the cleanup handler. This handlers need to stay valid even
+ *                 after the function has finished.
+ *
+ */
+
+void nrf_rpc_register_cleanup_handler(struct nrf_rpc_cleanup_handler * handler);
+
+
+/** @brief Temporairly suspend all RPC communication
+ *
+ * Calling this function automatically fails all communication until @ref nrf_rpc_resume is called.
+ * It also causes all pending request to fail immediately.
+ *
+ * @param cleanup set to true to also invoke all the custom cleanup handlers.
+ *
+ * @note If the cleanup parameter is set to true it may be needed to reset the RPC server before invoking
```

Comment:
```suggestion
 * @note If the cleanup parameter is set to true it may be needed to reset the peer before invoking
```
nRF RPC has no concept of client/server.
I think this note may be a bit confusing as to why the peer must be reset though. Also the local device may not be capable of resetting the peer. This note might require some explanation for this reason.